### PR TITLE
Release notes v22.2.6

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-6525

### CODE REVIEW

File: _includes/releases/v22.2/v22.2.6.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+## v22.2.6
+
+Release Date: March 3, 2023
+
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v22-2-6-security-updates">Security updates</h3>
+
+- Introduced the `server.user_login.downgrade_scram_stored_passwords_to_bcrypt.enabled` [cluster setting](../v22.2/cluster-settings.html) , which defaults to `true`. If it is `true` **and** `server.user_login.password_encryption` is `crdb-bcrypt`, then during login, the stored hashed password will be converted from [`SCRAM`](../v22.2/security-reference/scram-authentication.html) to `bcrypt`. Previously, the process to change passwords from `SCRAM` hashing to `bcrypt` hashing was always manual and took a long time. An operator may wish to use this cluster setting if their tools do not support `SCRAM`. [#97628][#97628]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:9]: nit:

>`server.user_login.password_encryption` is set to `crdb-bcrypt`

File: _includes/releases/v22.2/v22.2.6.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+<h3 id="v22-2-6-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- Fixed a bug in changefeeds where long running [initial scans](../v22.2/create-changefeed.html#initial-scan) will fail to generate checkpoints. Failure to generate [checkpoints](../v22.2/change-data-capture-overview.html#how-does-an-enterprise-changefeed-work) is particularly bad if the changefeed restarts. Without checkpoints, the changefeed will restart from the beginning, and in the worst case, when exporting substantially sized tables, the changefeed initial scan may not complete. [#97049][#97049]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:13]: supernit: `will fail` --> `would fail`

File: _includes/releases/v22.2/v22.2.6.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+- Added support for an optional external ID when [assuming a role](../v22.2/cloud-storage-authentication.html). You can use this by extending the `ASSUME_ROLE` parameter to the format `ASSUME_ROLE={role};external_is={id}`. When using [role chaining](../v22.2/cloud-storage-authentication.html#amazon-s3-role-chaining), you can associate each role in the chain with a different external ID. For example:
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:14]: correction: `external_id`

File: _includes/releases/v22.2/v22.2.6.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+    ~~~sql
+    ASSUME_ROLE={roleA};external_id={idA},{roleB};external_id=<idB>,{roleC}
+    ~~~
+
+    This will use external ID `{idA}` to assume delegate `{roleA}`, then use external ID `{idB}` to assume delegate `{roleB}`, and finally no external ID to assume the final role `{roleC}`. [#96531][#96531]
+- Changefeeds no longer require the `COCKROACH_EXPERIMENTAL_ENABLE_PER_CHANGEFEED_METRICS` environment variable in order to use [`metrics_label`](../v22.2/monitor-and-debug-changefeeds.html#using-changefeed-metrics-labels) option. [#97509][#97509]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:21]: supernit: `in order to use the ... option`

File: _includes/releases/v22.2/v22.2.6.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+- Added a hard limit of how much data can be flushed to system tables for SQL stats. [#97399][#97399]
+- Added support for expressions of the form `COLLATE "default"`, `COLLATE "C"`, and `COLLATE "POSIX"`. Since the default [collation](../v22.2/collate.html) cannot be changed currently, these expressions are all equivalent. The expressions are evaluated by treating the input as a normal string, and ignoring the collation. This means that comparisons between strings and collated strings that use `"default"`, `"C"`, or `"POSIX"` are now supported. Creating a column with the `"C"` or `"POSIX"` collations is still not supported. [#97415][#97415]
+- Previously, you could use [user-defined function](../v22.2/user-defined-functions.html) usage from tables with `SET DEFAULT` and `SET ON UPDATE` even though they are disallowed from `CREATE TABLE` and `ADD COLUMN`. This patch disallows those two cases from `ALTER TABLE ... ALTER COLUMN`. [#97430][#97430]
+- Increased the default value of `sql.stats.cleanup.rows_to_delete_per_txn` to 10k, to increase efficiency of the cleanup job for SQL statistics. [#97724][#97724]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:30]: nit: `10k` --> `10000`

File: _includes/releases/v22.2/v22.2.6.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+<h3 id="v22-2-6-operational-changes">Operational changes</h3>
+
+- A [`BACKUP`](../v22.2/backup.html) that encounters too many retryable errors will now fail instead of pausing to allow subsequent backups the chance to succeed. [#96716][#96716]
+
+<h3 id="v22-2-6-db-console-changes">DB Console changes</h3>
+
+- Previously, when the SQL API returned a "max size reached" error, the DB Console was only showing the error, but not the data that was also being returned. Now, the Statement Insights, Transaction Insights, and Schema Insights pages also show the data. [#97470][#97470]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:38]: nit: `was only showing` --> `would only show`

Comment (andf-crl) [reply _includes/releases/v22.2/v22.2.6.md:38]: nit: emphasized: `**pagename**` formatting for the three page names. No need to link.

File: _includes/releases/v22.2/v22.2.6.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+- Added the following new metrics to track memory usage of prepared statements in sessions:
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:39]: I might just say "Added 22 new metrics to track ... in sessions,." and leave it at that. Up to you.

File: _includes/releases/v22.2/v22.2.6.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+- Removed the following log message that was produced frequently: `lease [...] expired before being followed by lease [...]; foreground traffic may have been impacted`. [#97377][#97377]
+- Fixed a bug in the query engine that could cause incorrect results in some cases when a [zigzag join](../v22.2/joins.html) was planned. The bug could occur when the two indexes used for the zigzag join had a suffix of matching columns, but with different directions. For example, planning a zigzag join with `INDEX(a ASC, b ASC)` and `INDEX(c ASC, b DESC)` could cause incorrect results. This bug has existed since at least v19.1. It is now fixed, because the [optimizer](../v22.2/cost-based-optimizer.html) will no longer plan a zigzag join in such cases. [#97441][#97441]
+- Fixed a bug in [`ALTER TABLE ... ADD COLUMN`](../v22.2/alter-table.html#add-column) when the new column name requires quoting due to mixed case or special characters and the statement is not run in an explicit or multi-statement transaction. [#97568][#97568]
+- Fixed a bug when formatting create statements for user-defined types that require quoting, which might prevent those statements from round-tripping. [#97568][#97568]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:77]: This is my favorite malapropic verbification from upstream yet: "round-tripping!" No action, from docs, needed.

File: _includes/releases/v22.2/v22.2.6.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+- Added support for disabling cross-descriptor validation on lease renewal, which can be problematic when there are lots of descriptors with lots of foreign key references. In these cases, the cross-reference validation could block schema changes. This can be enabled with `sql.catalog.descriptor_lease_renewal_cross_validation`. [#97635][#97635]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:78]: Though: `lots` of `lots` almost as good though! Again, no action from docs.

File: _includes/releases/v22.2/v22.2.6.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+- Fixed a bug where a [backup](../v22.2/take-backups-with-revision-history-and-restore-from-a-point-in-time.html) of keys with many revisions would fail with `pebble: keys must be added in order`. [#97062][#97062]
+- Previously, `ALTER TABLE ... INJECT STATISTICS` command would fail if a column with `COLLATED STRING` type had histograms to be injected. This is now fixed. The bug has been present since at least v21.2. [#97491][#97491]
+
+<h3 id="v22-2-6-performance-improvements">Performance improvements</h3>
+
+- Long chains of [incremental backups](../v22.2/take-full-and-incremental-backups.html) and restore of such chains will now allocate less memory during the un-marshaling of metadata. [#97210][#97210]
```
Comment (andf-crl) [inline _includes/releases/v22.2/v22.2.6.md:87]: - nit :`restores of such chains`

**Review by andf-crl — APPROVED** [review]
LGTM, just a few tiny language recommendations. Thanks!

File: _includes/releases/v22.2/v22.2.6.md (hunk 1 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: Done!

File: _includes/releases/v22.2/v22.2.6.md (hunk 2 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: Check!

File: _includes/releases/v22.2/v22.2.6.md (hunk 3 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: Oooops

File: _includes/releases/v22.2/v22.2.6.md (hunk 4 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: yesss!

File: _includes/releases/v22.2/v22.2.6.md (hunk 5 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: Done!

File: _includes/releases/v22.2/v22.2.6.md (hunk 6 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: Thanks!

File: _includes/releases/v22.2/v22.2.6.md (hunk 8 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md:55]: LOL

File: _includes/releases/v22.2/v22.2.6.md (hunk 10 above)
Comment (kathancox) [reply _includes/releases/v22.2/v22.2.6.md]: Yepp!

**Pull request merged**